server_admin/topics/authentication/x509.adoc

```diff
@@ -209,6 +209,61 @@ image:images/x509-directgrant-flow.png[]
+
+However when {project_name} listens to HTTP requests behind the loadbalancer or reverse proxy, it might be the proxy server, which extracts the client certificate and establish the mutual SSL connection. Reverse proxy usually put the authenticated client certificate to the HTTP header of the underlying request and forwards it to the backend {project_name} server. In this case, {project_name} must be able to lookup the X.509 certificate chain from the HTTP headers instead of from the attribute of HTTP request like done for Undertow.
+
+If {project_name} is behind the reverse proxy, you usually need to configure alternative provider of the `x509cert-lookup` SPI in `{project_dirref}/standalone/configuration/standalone.xml` file. Besides the `default` provider, which lookups the certificate from the HTTP header, we have 2 builtin providers: `haproxy` and `apache`.
+
+===== HAProxy certificate lookup provider
+
+This provider can be used when your {project_name} server is behind the HAProxy reverse proxy. The provider needs to be configured like this:
+
+[source,xml]
+----
+<spi name="x509cert-lookup">
+    <default-provider>haproxy</default-provider>
+    <provider name="haproxy" enabled="true">
+        <properties>
+            <property name="sslClientCert" value="SSL_CLIENT_CERT"/>
+            <property name="sslCertChainPrefix" value="CERT_CHAIN"/>
+            <property name="certificateChainLength" value="10"/>
+        </properties>
+    </provider>
+</spi>
+----
+
+In this example configuration, the client certificate will be lookup from the HTTP header `SSL_CLIENT_CERT` and the other certificates from it's chain will be lookup from the HTTP headers like `CERT_CHAIN_0` , `CERT_CHAIN_1`, ..., `CERT_CHAIN_9` . The attribute `certificateChainLength` is the maximum length of the chain, so the last one tried attribute would be `CERT_CHAIN_9` .
+
+Consult the HAProxy documentation for the details how the HTTP Headers for the client certificate and client certificate chain can be configured and what are their proper names.
+
+===== Apache certificate lookup provider
+
+This provider can be used when your {project_name} server is behind the Apache reverse proxy. The provider needs to be configured like this:
+
+[source,xml]
+----
+<spi name="x509cert-lookup">
+    <default-provider>apache</default-provider>
+    <provider name="apache" enabled="true">
+        <properties>
+            <property name="sslClientCert" value="SSL_CLIENT_CERT"/>
+            <property name="sslCertChainPrefix" value="CERT_CHAIN"/>
+            <property name="certificateChainLength" value="10"/>
+        </properties>
+    </provider>
+</spi>
+----
+
+The configuration is same like for `haproxy` provider. Again, consult the Apache documentation for the details how the HTTP Headers for the client certificate and client certificate chain can be configured and what are their names.
```

Comment:
The configuration is same as for the `haproxy` provider. Consult the Apache documentation for the details of how the HTTP Headers for the client certificate and client certificate chain can be configured and their proper names.